client/src/components/room/roomEntry.tsx

```diff
@@ -0,0 +1,200 @@
+  deleteRoom: (id: number) => void;
+}
+
+export function RoomEntry(
+  props: RoomEntryProps & RoomEntryDispatch
+): JSX.Element {
+  const [deleteDialogOpen, setDeleteDialogOpen] = React.useState(false);
+  const [editDialogOpen, setEditDialogOpen] = React.useState(false);
+  const classes = useStyles();
+
+  function handleDeleteClose() {
+    setDeleteDialogOpen(false);
+  }
+
+  function handleEditClose() {
+    setEditDialogOpen(false);
+  }
+
+  return (
+    <React.Fragment>
+      <Accordion>
+        <AccordionSummary
+          expandIcon={<ExpandMore />}
+          aria-controls="panel1a-content"
+          id="panel1a-header"
+        >
+          <div className={classes.roomHeading}>
+            <Fade
+              in={props.editMode === EditMode.edit}
+              mountOnEnter
+              unmountOnExit
+            >
+              <Button
+                className={classes.leftButton}
+                color="secondary"
+                onClick={(e) => {
+                  e.stopPropagation();
+                  setDeleteDialogOpen(true);
+                }}
+              >
+                <Delete />
+              </Button>
+            </Fade>
+            {props.editMode === EditMode.view && (
+              <Switch
+                className={classes.leftButton}
+                color="default"
+                onClick={(e) => {
+                  e.stopPropagation();
+                }}
+                onChange={() => {
+                  // TODO
+                }}
+              />
+            )}
+            <Typography className={classes.roomTitle}>
+              {props.room.name}
+            </Typography>
+            <Fade
+              in={props.editMode === EditMode.edit}
+              mountOnEnter
+              unmountOnExit
+            >
+              <Button
+                className={classes.rightButton}
+                onClick={(e) => {
+                  e.stopPropagation();
+                  setEditDialogOpen(true);
+                }}
+              >
+                <Edit />
+              </Button>
+            </Fade>
+          </div>
+        </AccordionSummary>
+        <AccordionDetails className={classes.detailContainer}>
+          <Typography>TODO: scenes for {props.room.name}</Typography>
+        </AccordionDetails>
+      </Accordion>
+
+      <InputDialog
+        onConfirm={(name) => {
+          handleEditClose();
+          props.editRoom({
+            ...props.room,
+            name
+          });
+        }}
+        onCancel={handleEditClose}
+        open={editDialogOpen}
+        title="Edit Room"
+        description='Enter a descriptive name for the room you wish to add. A room in
+          Home Lights represents a physical room in your home, e.g.
+          "kitchen," "guest bedroom", etc. The room name
+          must not already be in use.'
+        inputTitle="Room Name"
+        inputPlaceholder="e.g. Kitchen"
+      />
+
+      <ConfirmDialog
```

Comment:
Let's pull these components out:
- Power
- Edit
- Delete

Edit and Delete should have their confirm/inputs

Reply:
Let's put the buttons with their modals.